src/env.sh

```diff
@@ -57,33 +51,25 @@ env_init_pkg() {
     # Check if the package exists/has an ellipsis.sh file
     if [ -f "$PKG_PATH/ellipsis.sh" ]; then
         # Check if the package has an init hook
-        grep '^pkg.init' "$PKG_PATH/ellipsis.sh" 2>&1 >/dev/null
-        if [ $? -eq 0 ]; then
-            # Use some magic to extract the init function
-            pkg_init="$(bash -c "source $PKG_PATH/ellipsis.sh; declare -f pkg.init")"
-            pkg_init="$(echo "$pkg_init" | sed 's/pkg.init/pkg_init/')"
+        pkg_init="$(sed -n '/^pkg.init\(\)/,/^}/p' "$PKG_PATH/ellipsis.sh" | sed '1 s/\./_/')"
+        if [ ! -z "$pkg_init" ]; then
             eval "$pkg_init"
-
             # Handle some edge cases by checking if the function is available
-            command -v 'pkg_init' 2>&1 >/dev/null
-            if [ $? -eq 0 ]; then
-                cwd="$(pwd)"
-                cd "$PKG_PATH"
-
-                # Init or log an error
-                if ! pkg_init; then
-                    env_api log.error "Ellipsis could not initialize $PKG_NAME"
-                fi
-
-                cd "$cwd"
```

Comment:
We should still cd to the original working dir after initializing the last package, as our terminal wil otherwise open in the `$PKG_PATH` of the last initialized package.

But it might be better to do this after the for loop that inits the packages instead of after each package init.

Reply:
This should be fixed